app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,158 @@
+from django.test import TestCase
+
+from rest_framework import status
+from rest_framework.test import APIClient
+from rest_framework.utils import json
+
+from core.models import Recipe, Ingredient
+from recipe.serializers import RecipeSerializer
+
+
+def sample_ingredient(name='Cinnamon', recipe={'name': 'Some name recipe'}):
+    """Create and return a sample ingredient"""
+    return Ingredient.objects.create(name=name, recipe=recipe)
+
+
+def sample_recipe(**params):
+    """Create and return a sample recipe"""
+    defaults = {
+        'name': 'Pizza',
+        'description': 'Put in the ove',
+    }
+    defaults.update(params)
+
+    return Recipe.objects.create(**defaults)
+
+
+class RecipeApiTests(TestCase):
+    """Test unauthenticated recipe API access"""
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_create_basic_recipe(self):
+        """Test POST create recipe"""
+        payload = {
+            'name': 'Pizza',
+            'description': 'Put in the ove',
+            'ingredients': [
+                {'name': 'dough'},
+                {'name': 'cheese'},
+                {'name': 'tomato'}
+            ]
+        }
+        res = self.client.post('/recipes/',
+                               content_type='application/json',
+                               data=json.dumps(payload))
+
+        self.assertEqual(res.status_code, status.HTTP_201_CREATED)
+
+        all_recipes = Recipe.objects.all()
+        self.assertEqual(len(all_recipes), 1)
+
+        recipe = Recipe.objects.get(id=res.data['id'])
+        self.assertEqual(payload['name'], getattr(recipe, 'name'))
+        self.assertEqual(payload['description'],
+                         getattr(recipe, 'description'))
+
+    def test_retrieve_recipes(self):
+        """Test GET get all recipes /recipe/"""
+        recipe1 = sample_recipe()
+        recipe2 = sample_recipe()
+        sample_ingredient(name='cheese', recipe=recipe1)
+        sample_ingredient(name='tomato', recipe=recipe1)
+
+        res = self.client.get('/recipes/', format="json")
+
+        recipes = Recipe.objects.all().order_by('-id')
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEquals(len(res.data), 2)
+
+        self.assertIn(recipe1, recipes)
+        self.assertIn(recipe2, recipes)
+
+        serializer1 = RecipeSerializer(recipe1)
+        serializer2 = RecipeSerializer(recipe2)
+        self.assertIn(serializer1.data, res.data)
```

Comment:
It is OK to test the response payload like this comparing with the serializer output, but keep in mind that if the serializer is wrong you won't catch it. So depending on what you want to test (e.g. test that a new field is present) probably you want to test explicitly for content instead of testing that the response payload equals what the serializer produces.